### ISSUE TITLE
different render between 0.4.0 and 0.5.0 

### ISSUE THREAD

**Issue by vincentbonnetcg** [facebookresearch/pytorch3d#838]
Hello, I recently moved to PyTorch v0.5.0 and found a render difference when compared with PyTorch v0.4.0.

## 1. Instructions To Reproduce the Issue

The code below loads a simple geometry and renders it. The code produces an output image of size 300x512.

```
import os
import numpy
import torch
import matplotlib.pyplot as plt

# USER PARAMETERS
IMAGE_HEIGHT = 300
IMAGE_WIDTH = 512
IMAGE_RATIO = IMAGE_WIDTH / IMAGE_HEIGHT

# Data structures and functions for rendering
from pytorch3d.io import load_objs_as_meshes, load_obj
from pytorch3d.renderer import (
    BlendParams,
    look_at_view_transform,
    FoVPerspectiveCameras,
    PointLights,
    RasterizationSettings,
    MeshRenderer,
    MeshRasterizer,
    SoftPhongShader,
)
import urllib.request

# Download cow
urls = ['https://dl.fbaipublicfiles.com/pytorch3d/data/cow_mesh/cow.obj',
        'https://dl.fbaipublicfiles.com/pytorch3d/data/cow_mesh/cow.mtl',
        'https://dl.fbaipublicfiles.com/pytorch3d/data/cow_mesh/cow_texture.png']

names = ['cow.obj', 'cow.mtl', 'cow_texture.png']
for i in range(3):
    if os.path.exists(names[i]) == False:
        urllib.request.urlretrieve(urls[i], names[i])

# Create device
device = torch.device("cuda:0")

# Load mesh
mesh = load_objs_as_meshes(["cow.obj"], device=device)

# Initialize camera.
R, T = look_at_view_transform(2.7, 0, 180)
cameras = FoVPerspectiveCameras(device=device, R=R, T=T, aspect_ratio=IMAGE_RATIO)

# Initialize render
raster_settings = RasterizationSettings(
    image_size=(IMAGE_HEIGHT, IMAGE_WIDTH),
    blur_radius=0.0,
    faces_per_pixel=1,
)
lights = PointLights(device=device, location=[[0.0, 0.0, -3.0]])
rasterizer=MeshRasterizer(
    cameras=cameras,
    raster_settings=raster_settings
)
blend_params = BlendParams(sigma=1e-4, gamma=1e-4, background_color=(0.0, 0.0, 0.0))
shader = SoftPhongShader(
        device=device,
        cameras=cameras,
        blend_params=blend_params,
        lights=lights)

renderer = MeshRenderer(
    rasterizer, shader)

# Render image
images = renderer(mesh)
plt.figure(figsize=(10, 10))
plt.imshow(images[0, ..., :3].cpu().numpy())
plt.axis("off");
```
## 2. Result 
As you can see, the same code produces two different renders with two PyTorch versions.

![diff_pytorch_version](https://user-images.githubusercontent.com/13979887/133020503-b2f38e6d-3183-49fb-962b-29f974c72fec.png)

## 3. Questions
- Do you know why there is a difference between the two versions? 
- How can we achieve our original result with PyTorch v0.5.0? 

Thanks,
Vincent

**Comment by gkioxari** [comment]
Yes, we introduced a breaking change in 0.5.0 which we detail in the release notes. I have also posted a more detailed discussion here https://github.com/facebookresearch/pytorch3d/discussions/783

We have slightly changed the way we treat cameras which are now made more compatible with the rendering engine. The changes impact non square rasterization which is your case exactly so I suggest you read the note in the discussion above carefully and adjust your code. Sorry for the inconvenience :)
